ape_solidity/compiler.py

```diff
@@ -510,23 +514,84 @@ def get_standard_input_json_from_settings(
             if missing_sources := [
                 x for x in vers_settings["outputSelection"] if not (pm.path / x).is_file()
             ]:
-                # See if the missing sources are from dependencies (they likely are)
-                # and cater the error message accordingly.
-                if dependencies_needed := [x for x in missing_sources if str(x).startswith("@")]:
-                    # Missing dependencies. Should only get here if dependencies are found
-                    # in import-strs but are not installed anywhere (not in project or globally).
-                    missing_str = ", ".join(dependencies_needed)
-                    raise CompilerError(
-                        f"Missing required dependencies '{missing_str}'. "
-                        "Install them using `dependencies:` "
-                        "in an ape-config.yaml or using the `ape pm install` command."
-                    )
-
-                # Otherwise, we are missing project-level source files for some reason.
-                # This would only happen if the user passes in unexpected files outside
-                # of core.
-                missing_src_str = ", ".join(missing_sources)
-                raise CompilerError(f"Sources '{missing_src_str}' not found in '{pm.name}'.")
+                # Potentially remappings are missing or wrongfully-including a contracts key.
```

Comment:
- [ ] Make this happen when building the import map instead or else other problems will occur